source/common/router/rds_impl.cc

```diff
@@ -73,10 +73,10 @@ RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl(
   // In V2 we use a Subscription model where the fetch can happen via gRPC, REST, or
   // local filesystem. If the subscription happens via local filesystem (e.g xds_integration_test),
   // then there is no actual RDS server, and hence no RDS cluster name.
-  if (rds.has_config_source() && rds.config_source().has_api_config_source()) {
-    cluster_name_ = rds.config_source().api_config_source().cluster_names()[0];
+  if (rds.has_config_source()) {
+    config_source_ = MessageUtil::getJsonStringFromMessage(rds.config_source(), true);
   } else {
-    cluster_name_ = "NOT_USING_CLUSTER";
+    config_source_ = "{}";
```

Comment:
Sorry quick question: Is there a test that should have caught this (switching to `{}`)? Is it easy to add one?

Reply:
@mattklein123  there is no test. Will look in to adding it.  One more thing, /routes is not working for routes defined statically. It is only iterating through the RdsRouteConfigProviders. https://github.com/envoyproxy/envoy/blob/master/source/common/router/rds_impl.cc#L215. Also it does not store the StaticRouteConfigProviders any where. Is this intentional or a bug? @htuch your thoughts?

Reply:
and the else block changing config_source = {}, I think we can get rid of the else block completely because if rds is configured, with out config_source, EnvoyException will be thrown. And that is validated with this test https://github.com/envoyproxy/envoy/blob/master/test/common/router/rds_impl_test.cc#L126.

Reply:
> Also it does not store the StaticRouteConfigProviders any where. Is this intentional or a bug? 

It just wasn't implemented. In the context of https://github.com/envoyproxy/envoy/issues/2421 we will want to revisit this.

> and the else block changing config_source = {}, I think we can get rid of the else block completely because if rds is configured, with out config_source, EnvoyException will be thrown.

Yup, great catch. If the code can't be hit just delete it and remove the if statement entirely. Thanks!